### ISSUE TITLE
Bug with conditional attribute

### ISSUE THREAD

**Issue by whitecolor** [canjs/canjs#2077]
In 2.3 (it worked in 2.2):
```HTML
{{#if preview}}next-page="{nextPage}"{{/if}}
```

If `preview` is `true` `nextPage` attribute value is string value `{nextPage}`, not actually value of `nextPage` in parent scope.

Maybe some why to do it with new binding sintax?

test is comming.

**Comment by justinbmeyer** [comment]
You tried this with master?

**Comment by whitecolor** [comment]
Yes with yesterday master.

**Comment by whitecolor** [comment]
In this case:

```HTML
<some-comp {{#unless allowSome}}disabled{{/unless}}></some-comp>
```

if `disabled` attr is `htmlbool` in `some-comp` component, it event breaks with en erorr:

```
node_modules/can/view/stache/stache.js 301 Error: / is currently not supported within a tag.
```

**Comment by justinbmeyer** [comment]
@whitecolor I get no errors with:

```
test("conditional attributes (#2077)", function(){
			can.Component.extend({
				tag: 'some-comp',
				viewModel: {
					define: {
						disabled: {
							type: 'htmlbool'
						}
					}
				}
			});
			var template = can.stache("<some-comp {{#unless allowSome}}disabled{{/unless}}></some-comp>");
			
			template();
		});
```

Can you submit a test?

**Comment by justinbmeyer** [comment]
So the problem was that this was never working great to begin with.  However in 2.0, it was working on component initialization because attribute callbacks were called before tag callbacks.  

This meant that `next-page="{nextPage}"` was added to the element before the tag was created.  

Now this happens after.  

I'm trying to track down why we made this change, although I think it makes sense that components are invoked first.

**Comment by justinbmeyer** [comment]
This commit changed each attribute to be added to the front: https://github.com/canjs/canjs/commit/bdb06d6a92e4b8de3a59d897fba03808f71cf80d

This has the effect of making the tag run last because it will be added first.

This was added because bindings needed the view-model to be created before they ran. This is no longer the case because can-component sets up bindings itself.

So the hack solution is to reverse this.  This will bring back compatibility.

But the better solution is to be able to handle attributes being added and removed.  

The better solution isn't easy to write because binding split between components and can/view/bindings.

I might make this work with can/view/bindings without components first, and then make components do it and try to share as much code as possible.

**Comment by justinbmeyer** [comment]
@whitecolor to be clear ... the "good" short term fix for this is to have `nextPage` added at end of turn.  This is slightly different then the behavior in 2.2.  However, we can fix that later by having some sort of `can.view.tagStart()` and `can.view.tagEnd()`.  `tagStart` will be called before any `can.view.attr`, and `tagEnd` will be called after all `can.view.attr`.  

Having both will allow us to turn off all the view-binding `attrs` and then initialize the component once all the attributes have been written.

**Comment by whitecolor** [comment]
Ok thanks, for now I just got rid of such cases in the code.

**Comment by daffl** [comment]
Closed via #2089
